dev/reference/alert-rules.md

```diff
@@ -0,0 +1,1276 @@
+---
+title: TiDB Cluster Alert Rules
+summary: Learn the alert rules in a TiDB cluster.
+category: reference
+---
+
+<!-- markdownlint-disable MD024 -->
+
+# TiDB Cluster Alert Rules
+
+This document describes the alert rules for different components in a TiDB cluster, including the rule descriptions and solutions of the alert items in TiDB, TiKV, PD, TiDB Binlog, Node_exporter and Blackbox_exporter.
+
+According to the severity level, alert rules can be divided into three categories (from high to low): emergency-level, critical-level and warning-level.
+
+## TiDB alert rules
+
+This section gives the alert rules for the TiDB component.
+
+### Emergency-level alerts
+
+Emergency-level alerts are often caused by a service or node failure. Manual intervention is required immediately.
+
+#### `TiDB_schema_error`
+
+* Alert rule:
+
+    `increase(tidb_session_schema_lease_error_total{type="outdated"}[15m]) > 0`
+
+* Description:
+
+    The latest schema information is not reloaded in TiDB within one lease. When TiDB fails to continue providing services, an alert is triggered.
+
+* Solution:
+
+    It is often caused by an unavailable Region or a TiKV timeout. You need to locate the issue by checking the TiKV monitoring items.
+
+#### `TiDB_tikvclient_region_err_total`
+
+* Alert rule:
+
+    `increase(tidb_tikvclient_region_err_total[10m]) > 6000`
+
+* Description:
+
+    When TiDB accesses TiKV, a Region error occurs. When the error is reported over 6000 times in 10 minutes, an alert is triggered.
+
+* Solution:
+
+    View the monitoring status of TiKV.
+
+#### `TiDB_domain_load_schema_total`
+
+* Alert rule:
+
+    `increase(tidb_domain_load_schema_total{type="failed"}[10m]) > 10`
+
+* Description:
+
+    The total number of failures to reload the latest schema information in TiDB. If the reloading failure occurs over 10 times in 10 minutes, an alert is triggered.
+
+* Solution:
+
+    Same as [`TiDB_schema_error`](#-tidb-schema-error).
+
+#### `TiDB_monitor_keep_alive`
+
+* Alert rule:
+
+    `increase(tidb_monitor_keep_alive_total[10m]) < 100`
+
+* Description:
+
+    Indicates whether the TiDB process still exists. If the number of times for `tidb_monitor_keep_alive_total` increases less than 100 in 10 minutes, the TiDB process might already exit and an alert is triggered.
+
+* Solution:
+
+    * Check whether the TiDB process is out of memory.
+    * Check whether the machine has restarted.
+
+### Critical-level alerts
+
+For the critical-level alerts, a close watch on the abnormal metrics is required.
+
+#### `TiDB_server_panic_total`
+
+* Alert rule:
+
+    `increase(tidb_server_panic_total[10m]) > 0`
+
+* Description:
+
+    The number of the panicked TiDB threads. When a panic occurs, an alert is triggered. The thread is often recovered, otherwise, TiDB will frequently restart.
```

Comment:
```suggestion
    The number of panicked TiDB threads. When a panic occurs, an alert is triggered. Generally, the panicked thread is recovered. Otherwise, TiDB will frequently restart.
```